File: c++/p005.cpp
#include <string>
#include <algorithm>
#include <list>
#include <iostream>
#include <math.h>  

using namespace std;

int gcd(int a, int b) {
	while (b != 0) {
		int c = a % b;
		a = b;
		b = c;
	}
	return a;
}

int main() {
	int count = 1;
	for (int i = 1; i < 21; i++) {
		count *= floor(i / gcd(i, count));
	}
	cout << "Result: \n";
	cout << count;
	return count;
}

File: lua/p005.lua
function gcd(a, b) 
	while b ~= 0 do
		c = a % b
		a = b
		b = c
	end
	return a
end

function main() 
	count = 1
	for i = 1, 20, 1 do
		count = count * math.floor(i / gcd(i, count))
	end
	return count

end

print(main())